src/components/ProductDetail.jsx

```diff
@@ -0,0 +1,48 @@
+import Loading from './../components/Loading';
+import { useFetch } from './../JS/useFetch';
+import { useParams, useNavigate } from 'react-router-dom';
+
+
+const ProductDetail = () => {
+    const productID = Number(useParams().productID) || 0;
+
+    const {data: product, loading, error} = useFetch('products/' + productID);
+
+    const navigate = useNavigate();
+    const handleShowProducts = () => {
+        navigate('/products');
+    };
+    
+    if(error != undefined && error.length > 0) { return <p style={{ textAlign:'center'}}>An error occurres {error}</p>; };
```

Comment:
`if(error&& error.length > 0) {` is equivalent if you want to save some keystrokes.